### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by JacksonElia**
Now that we have a little more time, I want to improve the readme

### CODE REVIEW

**Comment by harshil21** [comment]
A Table of Contents would be nice. Also a [mermaid diagram](https://github.blog/developer-skills/github/include-diagrams-markdown-files-mermaid/) showing the general architecture of the code would be very useful for new people looking at the code (excellent task for chatgpt)

File: README.md [hunk 1]
```diff
@@ -1,67 +1,58 @@
...
-git clone https://github.com/yourusername/AirbrakesV2.git
-cd AirbrakesV2
-```
-
-#### Set up a virtual environment:
-
-```
-uv venv
-source .venv/bin/activate
+This project is for controlling our Air brakes system with the goal of making our rocket "hit" its target apogee. We have a Raspberry Pi 4 as the brains of our system which runs our code. It connects to a servo motor to control the extension of our air brakes and an [IMU](https://www.microstrain.com/inertial-sensors/3dm-cx5-25) (basically an altimeter, accelerometer, and gyroscope). The code follows the [finite state machine](https://www.tutorialspoint.com/design_pattern/state_pattern.htm) design pattern, using the [`AirbrakesContext`](https://github.com/NCSU-High-Powered-Rocketry-Club/AirbrakesV2/blob/main/airbrakes/airbrakes.py) to manage interactions between the states, hardware, logging, and data processing.
```
Comment (harshil21) [inline README.md:4]:
```suggestion
This project is for controlling our Air brakes system with the goal of making our rocket "hit" its target apogee. We have a Raspberry Pi 4 as the brains of our system which runs our code. It connects to a servo motor to control the extension of our air brakes and an [IMU](https://www.microstrain.com/inertial-sensors/3dm-cx5-15) (basically an altimeter, accelerometer, and gyroscope). The code follows the [finite state machine](https://www.tutorialspoint.com/design_pattern/state_pattern.htm) design pattern, using the [`AirbrakesContext`](https://github.com/NCSU-High-Powered-Rocketry-Club/AirbrakesV2/blob/main/airbrakes/airbrakes.py) to manage interactions between the states, hardware, logging, and data processing. 
```

We have cx-15. Might be worth updating the class docstring of `IMU` as well.

File: README.md [hunk 2]
```diff
@@ -1,67 +1,58 @@
...
-```
-
-#### Set up a virtual environment:
-
-```
-uv venv
-source .venv/bin/activate
+This project is for controlling our Air brakes system with the goal of making our rocket "hit" its target apogee. We have a Raspberry Pi 4 as the brains of our system which runs our code. It connects to a servo motor to control the extension of our air brakes and an [IMU](https://www.microstrain.com/inertial-sensors/3dm-cx5-25) (basically an altimeter, accelerometer, and gyroscope). The code follows the [finite state machine](https://www.tutorialspoint.com/design_pattern/state_pattern.htm) design pattern, using the [`AirbrakesContext`](https://github.com/NCSU-High-Powered-Rocketry-Club/AirbrakesV2/blob/main/airbrakes/airbrakes.py) to manage interactions between the states, hardware, logging, and data processing. 
+
+https://github.com/user-attachments/assets/0c72a9eb-0b15-4fbf-9e62-f6a69e5fadaa
```
Comment (harshil21) [inline README.md:6]: nice, I thought this would be too sensitive / confidential information lol

File: README.md [hunk 3]
```diff
@@ -1,67 +1,58 @@
-# AirbrakesV2
-
+# Air BrakesV2 🚀
```
Comment (harshil21) [inline README.md:1]:
```suggestion
# AirbrakesV2 🚀
```
nah one word is better

File: README.md [hunk 4]
```diff
@@ -1,67 +1,58 @@
...
 
 ## Overview
```
Comment (harshil21) [inline README.md:3]: we need a ToC here

File: README.md [hunk 5]
```diff
@@ -1,67 +1,58 @@
...
+4. **Controls the servo extension** based on the current state's instructions (e.g., extends air brakes to slow down the rocket)
+5. **Logs all data** from the IMU, Data Processor, Servo, and States
+```mermaid
+flowchart TD
+    A[Main Loop] --> B[Airbrakes Context]
+    B --> C[IMU - Get Data] --> B
+    B --> D[Data Processor - Process Data] --> B
+    B --> F[Update Current State] --> B
+    B --> G[Servo - Control Air Brakes] --> B
+    B --> E[Logger - Log Data]
```
Comment (harshil21) [inline README.md:25]: this is fine but I was picturing a more detailed diagram, showing the 2 child processes (IMU and logger) and then also including the other classes, like `IMUDataPacket`, `ProcessedDataPacket`, and `LoggedDataPacket`, `States`, etc. I can sketch a diagram during today's meeting and ask an llm to convert it to mermaid

File: README.md [hunk 6]
```diff
@@ -1,67 +1,58 @@
...
+2. Motor Burn - when the motor is burning and the rocket is accelerating
+3. Coast - when the motor has burned out and the rocket is coasting, this is when air brakes will be deployed
+4. Free Fall - when the rocket is falling back to the ground after apogee, this is when the air brakes will be
+retracted
+5. Landed - when the rocket has landed on the ground
+<img alt="graph" src="https://github.com/user-attachments/assets/39cf0556-d388-458b-8668-64177506c9de" width="70%">
 
+### File Structure
 
+We have put great care into keeping the file structure of this project organized and concise. Try to be intentional with where placing new files or directories.
```
Comment (harshil21) [inline README.md:41]:
```suggestion
We have put great effort into keeping the file structure of this project organized and concise. Try to be intentional with placing new files or directories.
```

File: README.md [hunk 7]
```diff
@@ -1,67 +1,58 @@
...
+# Air BrakesV2 🚀
 
 ## Overview
-This project controls the extension of airbrakes using a servo motor. It includes modules for logging data and managing the servo motor.
-
-### Installation
-
-#### Install uv:
-
-[uv](https://docs.astral.sh/uv/getting-started/installation/) is a python and project manager, which makes handling different python versions and virtual enviornments easier.
```
Comment (harshil21) [inline README.md:11]: no `uv` :(

File: README.md [hunk 8]
```diff
@@ -70,42 +61,98 @@ AirbrakesV2/
...
+
+### Set up a virtual environment:
+
+```bash
+python -m venv .venv
+
+# For Linux
+source .venv/bin/activate
+# For Windows
+venv\Scripts\activate
```
Comment (harshil21) [inline README.md:83]:
```suggestion
.\.venv\Scripts\activate
```
isn't it this?

File: README.md [hunk 9]
```diff
@@ -70,42 +61,98 @@ AirbrakesV2/
...
+```bash
+ssh pi@[IP.ADDRESS]
+# Its password is raspberry
+cd AirbrakesV2/
+```
 
-However during development, you may want to run individual scripts to test components. For example, to test the servo, run:
-```python3 -m scripts.run_servo```
+### Install and start the pigpio daemon on the Raspberry Pi:
+_Every time the pi boots up, you must run this in order for the servo to work. We have already added this command to run on startup, but you may want to confirm that it is running, e.g. by using `htop`._
```
Comment (harshil21) [inline README.md:135]:
```suggestion
_Every time the pi boots up, you must run this in order for the servo to work. We have already added this command to run on startup, but you may want to confirm that it is running, e.g. by using [`htop`](https://htop.dev/)._
```

File: README.md [hunk 10]
```diff
@@ -70,42 +61,98 @@ AirbrakesV2/
...
+# For Windows
+venv\Scripts\activate
+```
+
+### Install the required dependencies:
+
+```bash
+pip install .[dev]
+```
+_There are libraries that only fully work when running on the Pi (gpiozero, mscl), so if you're having trouble importing them locally, program the best you can and test your changes on the pi._
```
Comment (harshil21) [inline README.md:91]: we should mention `mscl` installation instructions.

**Review by harshil21 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (JacksonElia) [reply README.md:6]: Ehh I don't love the idea of keeping secrets. If someone directly tells us to close things down, we will, but until then I like keeping it open.

File: README.md (hunk 7 above)
Comment (JacksonElia) [reply README.md:11]: I thought about it more -- while it's a nice thing, I don't think it's required/related specifically to our project. If someone wants to use it, that's great, but I don't think we should make them do it -- just trying to simplify installation

File: README.md (hunk 10 above)
Comment (JacksonElia) [reply README.md:91]: I was thinking about that, but do you think it's necessary? It's already installed/setup on the pi, and having it installed on your laptop doesn't matter that much

Comment (harshil21) [reply README.md:91]: It's important for the future people who want to setup `mscl` on their Pi's, e.g. I had to do it when swapping the SD card

**Review by harshil21 — APPROVED** [review]
Architecture mermaid diagram changes can go into #36

**Pull request merged**